errors.go

```diff
@@ -0,0 +1,110 @@
+package netlink
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"os"
+)
+
+// Error messages which can be returned by Validate.
+var (
+	errMismatchedSequence = errors.New("mismatched sequence in netlink reply")
+	errMismatchedPID      = errors.New("mismatched PID in netlink reply")
+	errShortErrorMessage  = errors.New("not enough data for netlink error code")
+)
+
+// Errors which can be returned by a Socket that does not implement
+// all exposed methods of Conn.
+var (
+	errNotSupported = errors.New("operation not supported")
+)
+
+// notSupported provides a concise constructor for "not supported" errors.
+func notSupported(op string) *OpError {
+	return &OpError{
+		Op:  op,
+		Err: errNotSupported,
+	}
+}
+
+// IsNotExist determines if an error is produced as the result of querying some
+// file, object, resource, etc. which does not exist.  Users of this package
+// should always use netlink.IsNotExist, rather than os.IsNotExist, when
+// checking for specific netlink-related errors.
+//
+// Errors types created by this package, such as OpError, can be used with
+// IsNotExist, but this function also defers to the behavior of os.IsNotExist
+// for unrecognized error types.
+func IsNotExist(err error) bool {
+	switch err := err.(type) {
+	case *OpError:
+		// TODO(mdlayher): more error handling logic?
+
+		// Unwrap the inner error and use the stdlib's logic.
+		return os.IsNotExist(err.Err)
+	default:
+		return os.IsNotExist(err)
+	}
+}
+
+var _ error = &OpError{}
+var _ net.Error = &OpError{}
+
+// An OpError is an error produced as the result of a failed netlink operation.
+type OpError struct {
+	// Op is the operation which caused this OpError, such as "send"
+	// or "receive".
+	Op string
+
+	// Err is the underlying error which caused this OpError.
+	//
+	// If Err was produced by a system call error, Err will be of type
+	// *os.SyscallError. If Err was produced by an error code in a netlink
+	// message, Err will contain a raw error value type such as a unix.Errno.
+	//
+	// Most callers should inspect Err using a helper such as IsNotExist.
+	Err error
+}
+
+func (e *OpError) Error() string {
+	if e == nil {
+		return "<nil>"
+	}
+
+	return fmt.Sprintf("netlink %q: %v", e.Op, e.Err)
```

Comment:
Tiny nit: the `net` package does not quote the operation string: https://golang.org/src/net/net.go#L455. To me, `netlink send: something went wrong` looks better than `netlink "send": something went wrong` since I'm used to seeing errors from package `net`.